# Data Analysis with NumPy

![numpy.jpg](numpy.jpg)

## Importing Numpy and Pandas

In [2]:
import numpy as np
import pandas as pd

# Importing Data

In [4]:
data = 'data/forestfires.csv'

df = pd.read_csv(data)

In [5]:
df.head()

,X,Y,month,day,FFMC,DMC,DC,ISI,temp,RH,wind,rain,area
0,7,5,mar,fri,86.2,26.2,94.3,5.1,8.2,51,6.7,0.0,0.0
1,7,4,oct,tue,90.6,35.4,669.1,6.7,18.0,33,0.9,0.0,0.0
2,7,4,oct,sat,90.6,43.7,686.9,6.7,14.6,33,1.3,0.0,0.0
3,8,6,mar,fri,91.7,33.3,77.5,9.0,8.3,97,4.0,0.2,0.0
4,8,6,mar,sun,89.3,51.3,102.2,9.6,11.4,99,1.8,0.0,0.0


## Dataset Description

Attribute Information -

The dataset contains 517 instances and 13 attributes. The attribute information is as follows:-

* X - x-axis spatial coordinate within the Montesinho park map: 1 to 9
* Y - y-axis spatial coordinate within the Montesinho park map: 2 to 9
* month - month of the year: 'jan' to 'dec'
* day - day of the week: 'mon' to 'sun'
* FFMC - FFMC index from the FWI system: 18.7 to 96.20
* DMC - DMC index from the FWI system: 1.1 to 291.3
* DC - DC index from the FWI system: 7.9 to 860.6
* ISI - ISI index from the FWI system: 0.0 to 56.10
* temp - temperature in Celsius degrees: 2.2 to 33.30
* RH - relative humidity in %: 15.0 to 100
* wind - wind speed in km/h: 0.40 to 9.40
* rain - outside rain in mm/m2 : 0.0 to 6.4
* area - the burned area of the forest (in ha): 0.00 to 1090.84

In [7]:
fires = df.iloc[:5, 8:]
print(fires)

   temp  RH  wind  rain  area
0   8.2  51   6.7   0.0   0.0
1  18.0  33   0.9   0.0   0.0
2  14.6  33   1.3   0.0   0.0
3   8.3  97   4.0   0.2   0.0
4  11.4  99   1.8   0.0   0.0


# NumPy ndarray object

numpy.array(object, dtype=None, copy=True, order=None, subok=False, ndmin=0)

The above constructor takes the following parameters:
* object - Any object exposing the array interface method returns an array, or any (nested) sequence.
* dtype - Desired data type of array, optional.
* copy - (Optional) - By default (true), the object is copied.
* order - C (row major) or F (column major) or A (any) (default).
* subok - By default, returned array forced to be a base class array. If true, sub-classes passed through.
* ndmin - Specifies minimum dimensions of resultant array.

In [8]:
# one-dimensional array
a = np.array([1,2,3])
print(a)

[1 2 3]


In [11]:
n1 = np.array(fires[:1], dtype=np.float32)
print(n1)

[[ 8.2 51.   6.7  0.   0. ]]


In [12]:
# two-dimensional array
b = np.array([
    [1,2],
    [3,4]
])

print(b)

[[1 2]
 [3 4]]


In [13]:
n2 = np.array(fires[:2], dtype=int)

print(n2)

[[ 8 51  6  0  0]
 [18 33  0  0  0]]


In [14]:
# there-dimensional array
c = np.array([
    [1,2],
    [3,4],
    [5,6]
])

print(c)

[[1 2]
 [3 4]
 [5 6]]


In [15]:
n3 = np.array(fires[:3], dtype=np.int64)

print(n3)

[[ 8 51  6  0  0]
 [18 33  0  0  0]
 [14 33  1  0  0]]


In [21]:
# minimum dimensions

d = np.array([1,2,3,4,5], ndmin=2)
print(d)

[[1 2 3 4 5]]


In [22]:
# dtype parameter

e = np.array([1,2,3,4,5], dtype=complex)

print(e)

[1.+0.j 2.+0.j 3.+0.j 4.+0.j 5.+0.j]


## NumPy data types